### PR TITLE
Include JSON schema and Help() func in Plugin documentation

### PR DESCRIPTION
**Pull request by josefkarasek**
## Description

Changes proposed in this pull request:

- Add section for `JSON shcema`
- Add section for `Help()`

## Related issue(s)

https://github.com/kubeshop/botkube/issues/893

### CODE REVIEW

File: docs/plugin/custom-executor.md [hunk 1]
```diff
@@ -93,13 +113,28 @@ For a final implementation, see the [Botkube template repository](./quick-start.
...
 
+7. Add the required `Help` method:
+
+   ```go
+   // Help returns help message
+   func (*EchoExecutor) Help(ctx context.Context) (interactive.Message, error) {
+    return interactive.Message{
+      Base: interactive.Base{
+        Body: interactive.Body{
+          CodeBlock: "echo prints out given input string",
```
Comment (pkosiec) [inline docs/plugin/custom-executor.md:131]:
```suggestion
          CodeBlock: "Echo prints out given input string.",
```

File: docs/plugin/custom-executor.md [hunk 2]
```diff
@@ -77,10 +78,29 @@ For a final implementation, see the [Botkube template repository](./quick-start.
...
+              "type": "array",
+              "items": [
+                "type": "string",
+                "enum": ["bold", "italic"]
+              ]
+            }
+          },
+          "required": []
+        }`),
+       },
```
Comment (pkosiec) [inline docs/plugin/custom-executor.md:103]: Please use a real JSON schema:
https://github.com/kubeshop/botkube/blob/main/cmd/executor/echo/main.go#LL73C19-L73C19

File: docs/plugin/custom-executor.md [hunk 3]
```diff
@@ -93,13 +113,28 @@ For a final implementation, see the [Botkube template repository](./quick-start.
...
+      Base: interactive.Base{
+        Body: interactive.Body{
+          CodeBlock: "echo prints out given input string",
+        },
+      },
+    }, nil
+   }
+   ```
+
    The `Execute` method is the heart of your executor plugin. This method runs your business logic and returns the output as plain text. Next, the Botkube core sends back the response to a given communication platform.
```
Comment (pkosiec) [inline docs/plugin/custom-executor.md:138]: This is applicable for point no 6, so the new point no 7 is placed incorrectly. Please fix this.

File: docs/plugin/custom-executor.md (hunk 2 above)
Comment (josefkarasek) [reply docs/plugin/custom-executor.md:103]: Consistency: the example uses `formatOptions: [italic, bold]` . Using `changeResponseToUpperCase` can create confusion.

Comment (pkosiec) [reply docs/plugin/custom-executor.md]: Ah, you're right, sorry. Good catch.

In this case could you please bring back the previous schema? The previous one was invalid, but I fixed it so you can copy and paste the following one:

```json
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "botkube/echo",
  "description": "example echo plugin",
  "type": "object",
  "properties": {
    "formatOptions": {
      "description": "options to format echoed string",
      "type": "array",
      "items": {
        "type": "string",
        "enum": [
          "bold",
          "italic"
        ]
      }
    }
  },
  "additionalProperties": false,
  "required": []
}
```

Thanks!

**Review by pkosiec — APPROVED** [review]
LGTM, but please see my previous comment

**Pull request merged**